qiskit/converters/dag_to_dagcanonical.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Helper function for converting a dag to a dagcanonical"""
+from qiskit.dagcircuit.dagcanonical import DAGcanonical
+
+
+def dag_to_dagcanonical(dag):
+    """Build a ``QuantumCircuit`` object from a ``DAGCircuit``.
+
+    Args:
+        dag (DAGCircuit): the input dag.
+
+    Return:
+        DAGcanonical: the DAG representing the input circuit as a dag canonical.
+    """
+
+    dagcanonical = DAGcanonical()
+    DAGcanonical.name = dag.name
+
+    qregs = list(dag.qregs.values())
+    cregs = list(dag.cregs.values())
+
+    for register in qregs:
+        dagcanonical.add_qreg(register)
+
+    for register in cregs:
+        dagcanonical.add_creg(register)
+
+    for node in dag.topological_op_nodes():
+        # Get arguments for classical control (if any)
+        inst = node.op.copy()
+        inst.condition = node.condition
```

Comment:
do you only use the `condition` part of the inst here? If so then there are probably cheaper ways of getting that without doing a copy of the whole instruction (remember this does a deepcopy, and is called on every instruction of a large circuit). A condition is just a tuple of `(QuantumRegister, int)` which you can just create yourself.